metadata-ingestion/src/datahub/ingestion/source/mode.py

```diff
@@ -1566,6 +1604,34 @@ def emit_chart_mces(self) -> Iterable[MetadataWorkUnit]:
                             query_name=query["name"],
                         )
 
+    def emit_dataset_mces(self):
+        """
+        Emits MetadataChangeEvents (MCEs) for datasets within each space.
+        """
+        for space_token, _ in self.space_tokens.items():
+            datasets = self._get_datasets(space_token)
+
+            for report in datasets:
+                report_token = report.get("token", "")
+                queries = self._get_queries(report_token)
+                for query in queries:
+                    query_mcps = self.construct_query_or_dataset(
+                        report_token,
+                        query,
+                        space_token=space_token,
+                        report_info=report,
+                        is_mode_dataset=True,
+                    )
+                    chart_fields: Dict[str, SchemaFieldClass] = {}
+                    for wu in query_mcps:
+                        if isinstance(
+                            wu.metadata, MetadataChangeProposalWrapper
+                        ) and isinstance(wu.metadata.aspect, SchemaMetadataClass):
+                            schema_metadata = wu.metadata.aspect
+                            for field in schema_metadata.fields:
+                                chart_fields.setdefault(field.fieldPath, field)
```

Comment:
what's this code for?

Reply:
This code is to get all mode-datasets and emit into datahub datasets, as to show lineage and schema fields we need to find it query so need to traverse for query

Reply:
we dont need this code.